src/test-suites/utils.test.ts

```diff
@@ -0,0 +1,326 @@
+import * as core from '@actions/core'
+import os from 'os'
+import path from 'path'
+import fs from 'fs'
+import {v4} from 'uuid'
+import {getInputs, Inputs} from './utils'
+
+jest.mock('@actions/core')
+
+describe('Test Suites', () => {
+  describe('Utils', () => {
+    describe('getInputs', () => {
+      function setVariable(variable: string, value: string | undefined): void {
+        const originalValue = process.env[variable]
+
+        beforeEach(() => {
+          if (value === undefined) {
+            delete process.env[variable]
+          } else {
+            process.env[variable] = value
+          }
+        })
+
+        afterEach(() => {
+          process.env[variable] = originalValue
+        })
+      }
+
+      function createTempFile(contents: string): string {
+        const filepath = path.join(os.tmpdir(), v4())
+
+        beforeEach(() => {
+          fs.writeFileSync(filepath, contents)
+        })
+
+        afterEach(() => {
+          fs.rmSync(filepath)
+        })
+
+        return filepath
+      }
+
+      setVariable('GITHUB_ACTOR', 'actor')
+      setVariable('GITHUB_JOB', 'some-job')
+      setVariable('GITHUB_REF_NAME', 'some-ref-name')
+      setVariable('GITHUB_REPOSITORY', 'rwx-research/upload-captain-artifact')
+      setVariable('GITHUB_RUN_ATTEMPT', '4')
+      setVariable('GITHUB_RUN_ID', '1244592')
+      setVariable('GITHUB_SHA', 'some-commit-sha')
+      setVariable('GITHUB_TRIGGERING_ACTOR', undefined)
+
+      describe('when the payload is a pull request and there is an triggering actor', () => {
+        setVariable('GITHUB_TRIGGERING_ACTOR', 'triggering-actor')
+        setVariable(
+          'GITHUB_EVENT_PATH',
+          createTempFile(
+            JSON.stringify({
+              pull_request: {
+                head: {
+                  ref: 'some-branch'
+                }
+              }
+            })
+          )
+        )
```

Comment:
Long story short, it's near impossible to ergonomically mock the GitHub context since it's exporting a static object. Instead, we can just read the environment directly.